pkg/ccl/changefeedccl/alter_changefeed_stmt.go

```diff
@@ -268,6 +309,80 @@ func alterChangefeedPlanHook(
 	return fn, alterChangefeedHeader, false, nil
 }
 
+// setChangefeedTargets sets the appropriate target fields on the changefeed
+// statement based on the feed level. Database-level feeds get a DatabaseTarget,
+// while table-level feeds get TableTargets.
+func setChangefeedTargets(
+	ctx context.Context,
+	p sql.PlanHookState,
+	stmt *tree.CreateChangefeed,
+	prevDetails jobspb.ChangefeedDetails,
+	newTargets tree.ChangefeedTableTargets,
+) error {
+	if stmt.Level == tree.ChangefeedLevelDatabase {
+		targetSpec := prevDetails.TargetSpecifications[0]
+		txn := p.InternalSQLTxn()
+		databaseDescriptor, err := txn.Descriptors().ByIDWithLeased(txn.KV()).Get().Database(ctx, targetSpec.DescID)
+		if err != nil {
+			return err
+		}
+		stmt.DatabaseTarget = tree.ChangefeedDatabaseTarget(databaseDescriptor.GetName())
+	} else {
+		stmt.TableTargets = newTargets
+	}
+	return nil
+}
+
+// processFiltersForChangefeed validates and applies filter changes to the
+// changefeed statement. Filters are only supported for database-level feeds.
+func processFiltersForChangefeed(
+	stmt *tree.CreateChangefeed,
+	alterCmds tree.AlterChangefeedCmds,
+	prevDetails jobspb.ChangefeedDetails,
+) error {
+	filterCommands := generateNewFilters(alterCmds)
+
+	// Only DB-level feeds should have filters.
+	if len(filterCommands) > 0 && stmt.Level != tree.ChangefeedLevelDatabase {
+		return errors.Errorf("filters are only supported for database level changefeeds")
+	}
+
+	// We add the existing filter state to the changefeed statement for
+	// DB-level feeds. This ensures that existing filters are preserved when
+	// we alter options unrelated to filters and that we can error if we try
+	// to set both include and exclude filters at the same time.
+	if stmt.Level == tree.ChangefeedLevelDatabase {
+		targetSpec := prevDetails.TargetSpecifications[0]
+		filterOpt, err := parseFilterOptionFromTargetSpec(targetSpec)
+		if err != nil {
+			return err
+		}
+		stmt.FilterOption = filterOpt
+	}
+
+	// For table-level feeds, this should be a no-op since there are no
+	// filter commands as asserted above. Otherwise, apply each filter in order.
```

Comment:
nit: dont think this comment is necessary as we bail out at 346 if we're not a db level feed